tool/doc_packages.dart

```diff
@@ -216,8 +219,8 @@ Future<void> _exec(String command, List<String> args,
   });
 }
 
-bool _isOldSdkConstraint(var pubspecInfo) {
-  var environment = pubspecInfo['environment'];
+bool _isOldSdkConstraint(Map<String, dynamic> pubspecInfo) {
```

Comment:
is `Map<String, Object>` safe here?  Might as well go that route if we can.

Reply:
I'm not certain. YAML and JSON parsing are the only two places where I continue to use `dynamic` because (I think) `is` checks will fail if you use Object instead of dynamic.

It's hard for me to test this script. It consistently says `* failed *`...

Reply:
OK.  I think that `is` checks still work through Object (in fact, I think they're then required to disambiguate rather than calling members directly), but if this is too hard to test.... Well, I was about to say put in a TODO, but the use of dynamic is kind of an implicit TODO.

Reply:
Ah ok. Yeah I'm really not sure how to test this script.. :((

Agreed, any dynamic is its own TODO 🤣